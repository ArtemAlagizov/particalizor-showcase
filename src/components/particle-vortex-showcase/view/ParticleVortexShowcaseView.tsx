import React from "react";
import { IParticleVortexShowcaseView } from "./IParticleVortexShowcaseView";
import { ParticleVortex } from "particalizor-3000";
import { Styled } from "../styled";
import "antd/dist/antd.css";
import SliderWithLabel from "../../../ui-components/slider-with-label/container/SliderWithLabel";
import InputWithLabel from "../../../ui-components/input-with-label/container/InputWithLabel";

const ParticleVortexShowcaseView: React.FC<IParticleVortexShowcaseView> = ({
  onParticleNumberChange,
  onTraceWidthChange,
  onLifeTimeChange,
  onVortexNumberChange,
  onImageWidthChange,
  onImageHeightChange,
  onBackgroundColorChange,
  particleNumberValue,
  particleTraceWidthValue,
  particleLifeTimeValue,
  imageWidth,
  imageHeight,
  backgroundColor,
  vortexNumberValue
}) => {
  return (
    <Styled.ShowcaseComponentContainer>
      <Styled.PropertiesContainer>
        <Styled.PropertiesSet>
          <SliderWithLabel
            propertyName={"particleNumber"}
            selectedValue={particleNumberValue}
            onValueChange={onParticleNumberChange}
            label={"particle number"}
            range={{ min: 10, max: 1990 }}
            step={20}
          />
          <SliderWithLabel
            propertyName={"particleTraceWidthValue"}
            selectedValue={particleTraceWidthValue}
            onValueChange={onTraceWidthChange}
            label={"particle trace width"}
            range={{ min: 1, max: 800 }}
          />
          <SliderWithLabel
            propertyName={"particleLifeTimeValue"}
            selectedValue={particleLifeTimeValue}
            onValueChange={onLifeTimeChange}
            label={"particle life time"}
            range={{ min: 100, max: 7000 }}
            step={100}
          />
          <SliderWithLabel
            propertyName={"selectedVortexNumberValue"}
            selectedValue={vortexNumberValue}
            onValueChange={onVortexNumberChange}
            label={"vortex number"}
            range={{ min: 1, max: 20 }}
          />
        </Styled.PropertiesSet>
        <Styled.PropertiesSet>
          <SliderWithLabel
            propertyName={"imageWidth"}
            selectedValue={imageWidth}
            onValueChange={onImageWidthChange}
            label={"image width"}
            range={{ min: 200, max: 1200 }}
            step={100}
          />
          <SliderWithLabel
            propertyName={"imageHeight"}
            selectedValue={imageHeight}
            onValueChange={onImageHeightChange}
            label={"image height"}
            range={{ min: 200, max: 1200 }}
            step={100}
          />
          <InputWithLabel
            propertyName={"backgroundColor"}
            defaultValue={"#33344c"}
            label={"background color"}
            onValueChange={onBackgroundColorChange}
            selectedValue={backgroundColor}
          />
        </Styled.PropertiesSet>
      </Styled.PropertiesContainer>
      <Styled.Images>
        <Styled.ImageContainer>
          <ParticleVortex
            imageWidth={imageWidth}
            imageHeight={imageHeight}
            vortexNumber={vortexNumberValue}
            particleTraceWidth={particleTraceWidthValue}
            particleNumber={particleNumberValue}
            particleLifeTime={particleLifeTimeValue}
            backgroundColor={backgroundColor}
          />
        </Styled.ImageContainer>
      </Styled.Images>
    </Styled.ShowcaseComponentContainer>
  );
};

export default ParticleVortexShowcaseView;
