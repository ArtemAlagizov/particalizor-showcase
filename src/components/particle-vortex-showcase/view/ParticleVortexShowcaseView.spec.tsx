import React from "react";
import { shallow } from "enzyme";
import ParticleVortexShowcaseView from "./ParticleVortexShowcaseView";

describe("Given a ParticleVortexShowcaseView", () => {
  describe("when rendering", () => {
    it("should match snapshot", () => {
      const component = shallow(
        <ParticleVortexShowcaseView
          particleTraceWidthValue={1}
          particleNumberValue={1}
          particleLifeTimeValue={1}
          onTraceWidthChange={jest.fn()}
          onParticleNumberChange={jest.fn()}
          onLifeTimeChange={jest.fn()}
          imageHeight={100}
          imageWidth={100}
          backgroundColor={"red"}
          onBackgroundColorChange={jest.fn()}
          onImageHeightChange={jest.fn()}
          onImageWidthChange={jest.fn()}
          onVortexNumberChange={jest.fn()}
          vortexNumberValue={3}
        />
      );

      expect(component).toMatchSnapshot();
    });
  });
});
