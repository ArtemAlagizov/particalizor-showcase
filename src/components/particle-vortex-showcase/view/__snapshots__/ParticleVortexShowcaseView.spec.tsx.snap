// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a ParticleVortexShowcaseView when rendering should match snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <SliderWithLabel
        label="particle number"
        onValueChange={[MockFunction]}
        propertyName="particleNumber"
        range={
          Object {
            "max": 1990,
            "min": 10,
          }
        }
        selectedValue={1}
        step={20}
      />
      <SliderWithLabel
        label="particle trace width"
        onValueChange={[MockFunction]}
        propertyName="particleTraceWidthValue"
        range={
          Object {
            "max": 800,
            "min": 1,
          }
        }
        selectedValue={1}
      />
      <SliderWithLabel
        label="particle life time"
        onValueChange={[MockFunction]}
        propertyName="particleLifeTimeValue"
        range={
          Object {
            "max": 7000,
            "min": 100,
          }
        }
        selectedValue={1}
        step={100}
      />
      <SliderWithLabel
        label="vortex number"
        onValueChange={[MockFunction]}
        propertyName="selectedVortexNumberValue"
        range={
          Object {
            "max": 20,
            "min": 1,
          }
        }
        selectedValue={3}
      />
    </styled.div>
    <styled.div>
      <SliderWithLabel
        label="image width"
        onValueChange={[MockFunction]}
        propertyName="imageWidth"
        range={
          Object {
            "max": 1200,
            "min": 200,
          }
        }
        selectedValue={100}
        step={100}
      />
      <SliderWithLabel
        label="image height"
        onValueChange={[MockFunction]}
        propertyName="imageHeight"
        range={
          Object {
            "max": 1200,
            "min": 200,
          }
        }
        selectedValue={100}
        step={100}
      />
      <InputWithLabel
        defaultValue="#33344c"
        label="background color"
        onValueChange={[MockFunction]}
        propertyName="backgroundColor"
        selectedValue="red"
      />
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <Component
        backgroundColor="red"
        imageHeight={100}
        imageWidth={100}
        particleLifeTime={1}
        particleNumber={1}
        particleTraceWidth={1}
        vortexNumber={3}
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
